Automattic/wp-calypso
# Plans 2023: Migrate componentDidMount & componentDidUpdate logic to PlansFeaturesMain

Fixes https://github.com/Automattic/wp-calypso/issues/78993

## Proposed Changes

* Moves the lifecycle methods from `plan-features-2023-grid` to `plans-features-main`, uncoupling a couple of local dependencies
* Cleans up `plans-features-main` a little, removing the so-far temporary `OnboardingPricingGrid2023` component (we had created this previously for moving some parts into a function component). It is no longer necessary, and for the most just adds complexity with needing to pass props around
* Also removes the `AsyncLoad` component. We can reintroduce this later if necessary, although it will have a different form with individual components imported from `plan-features-2023-grid` next (per https://github.com/Automattic/wp-calypso/issues/78266). This was also necessary for now to pass a ref for the comparison-grid from `plans-features-main` (another temporary thing).

## Testing Instructions



* Ensure general sanity of `/plans[ paid | free ]`. That notices show properly, plan type selector works, etc. 
* Ensure general sanity on `/start` and any of the tailored flows under `/setup` (link-in-bio, newsletter)
* Ensure comparison grid scrolls smoothly into view when clicking on `compare plans`. Test this over the live image and double-check with production (or `trunk` in local) when in doubt
* Ensure the event `calypso_wp_plans_test_view` fires off once when visiting `/plans` or `/start`. It should be identical to production. 
    * from browser's dev tools: check Network tab for a `pixel.wp.com/t.gif` GET request with `_en: calypso_wp_plans_test_view` in payload/params

## Pre-merge Checklist
